src/core/helpers/ObjectHash.ts

```diff
@@ -1,7 +1,48 @@
 import * as crypto from 'crypto-js';
+import { HashableObjectType } from '../../api/enums/HashableObjectType';
+
 
 export class ObjectHash {
-    public static getHash(obj: any): string {
-        return crypto.SHA256(JSON.stringify(obj).split('').sort().toString()).toString();
+    public static async getHash(obj: any, type?: HashableObjectType): Promise<string> {
+        let revisedObj;
+        switch (type) {
+            case HashableObjectType.LISTINGITEM: {
+                const itemInformation = obj.ItemInformation;
+                const paymentInformation = obj.PaymentInformation;
+                const newObject = {
+                    ItemInformation: {
```

Comment:
the Message object does not contain any id's, and we should be able to use the message object to generate the hash to make sure the received object was not tampered with, so to clean up this code, you could use the factory to generate a message and then create the hash from that message.

now if we add some data to the model, the hash generation will not work correctly unless and until we add the new data in here too. if we add data to the model, we propably change the message too, and if we use the message here, we dont need to worry about updating this code in that situation.